src/test/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/AlkylStructureFragmenterTest.java

```diff
@@ -0,0 +1,431 @@
+            System.out.println("-----");
+            i++;
+        }
+    }
+    //</editor-fold>
+
+    //<editor-fold desc="Private Methods">
+
+    /**
+     * Compares two provided lists on equality while ignoring the lists' orders.
+     *
+     * @param aList1 First given list to compare
+     * @param aList2 Second given list to compare
+     * @return boolean whether given lists are equal
+     */
+    private boolean compareListsIgnoringOrder(ArrayList aList1, ArrayList aList2) {
+        if (aList1 == null || aList2 == null) {
+            return false;
+        }
+        if (aList1.size() != aList2.size()) {
+            return false;
+        }
+        for (Object o : aList1) {
+            aList2.remove(o);
+        }
+        return aList2.isEmpty();
+    }
+
+    /**
+     * Utility method generating SMILES notation strings for a given AtomContainerSet.
+     *
+     * @param anACSet given AtomContainerSet
+     * @return List containing the generated Strings
+     * @throws CDKException if SmilesGenerator is unable to generate String from structure
+     */
+    private List<String> generateSMILESFromACSet(IAtomContainerSet anACSet) throws CDKException {
+        List<String> tmpSmilesList = new ArrayList<>(anACSet.getAtomContainerCount());
+        for (IAtomContainer tmpAC : anACSet.atomContainers()) {
+            tmpSmilesList.add(ChemUtil.createUniqueSmiles(tmpAC));
+        }
+        return tmpSmilesList;
+    }
+
+    /**
+     * Private method to read a given structure file to a CDK atomcontainer set.
+     *
+     * @param aFileName Name of the file to read from
+     * @return IAtomContainerSet with the read structures as AtomContainers
+     * @throws FileNotFoundException if no file with the given name can be located
+     * @throws URISyntaxException if given name of file cannot be parsed as URI reference
+     */
+    private IAtomContainerSet readStructuresToACSet(String aFileName) throws FileNotFoundException, URISyntaxException {
+        URL tmpURL = this.getClass().getResource("/" + aFileName);
+        File tmpResourceFile = Paths.get(tmpURL.toURI()).toFile();
+        IteratingSDFReader tmpSDFReader = new IteratingSDFReader(new FileReader(tmpResourceFile), new SilentChemObjectBuilder());
+        AtomContainerSet tmpACSet = new AtomContainerSet();
+        String tmpIndexString = "ASFTest.AtomContainerIndex";
+        int tmpIndex = 0;
+        while (tmpSDFReader.hasNext()) {
+            IAtomContainer tmpAtomContainer = tmpSDFReader.next();
+            tmpAtomContainer.setProperty(tmpIndexString, tmpIndex);
+            try {
+                ChemUtil.saturateWithHydrogen(tmpAtomContainer);
+            } catch (CDKException aCDKException) {
+                throw new RuntimeException(aCDKException);
+            }
+            tmpACSet.addAtomContainer(tmpAtomContainer);
+            tmpIndex++;
+        }
+        return tmpACSet;
+    }
+    //</editor-fold>
+
```

Comment:
As stated in the comment above from the last review, this test class needs more step-by-step examples with actual molecules (check the SugarRemovalUtility test).